File: bot/core/models/observation.py
from typing import Optional
from pydantic import BaseModel
from .user import User
from .tg_geo import TgGeo
from .tg_media_group_item import TgMediaGroupItem

class Observation(BaseModel):
    id: str
    tgText: Optional[str]
    tgPhotoId: Optional[str]
    tgVideoId: Optional[str]
    tgVoiceId: Optional[str]
    tgDocumentId: Optional[str]
    tgVideoNoteId: Optional[str]
    tgGeo: Optional[TgGeo]
    tgMediaGroup: list[TgMediaGroupItem]
    user: User

File: packages/core-sdk/src/models/observation.ts
import { z } from "zod";
import { userSchema } from "./user.js";
import { tgGeoSchema } from "./tg-geo.js";
import { tgMediaGroupItemSchema } from "./tg-media-group-item.js";

export const observationSchema = z.object({
  id: z.string(),
  tgText: z.string().nullable(),
  tgPhotoId: z.string().nullable(),
  tgVideoId: z.string().nullable(),
  tgVoiceId: z.string().nullable(),
  tgDocumentId: z.string().nullable(),
  tgVideoNoteId: z.string().nullable(),
  tgGeo: tgGeoSchema.nullable(),
  tgMediaGroup: z.array(tgMediaGroupItemSchema),
  user: userSchema,
});

export type Observation = z.infer<typeof observationSchema>;
